contracts/FlowCallbackScheduler.cdc

```diff
@@ -307,13 +364,63 @@ access(all) contract FlowCallbackScheduler {
             self.historicStatusLimit = 30.0 * 24.0 * 60.0 * 60.0 // 30 days
         }
 
-        /// getNextID returns the next ID and increments the ID counter
-        access(self) fun getNextID(): UInt64 {
+        /// Borrows a reference to the specified callback
+        access(contract) view fun borrowCallback(id: UInt64): &CallbackData? {
+            return &self.callbacks[id]
+        }
+
+        /// calculate fee by converting execution effort to a fee in Flow tokens.
+        access(all) view fun calculateFee(executionEffort: UInt64, priority: Priority): UFix64 {
+            // Use the official FlowFees calculation
+            let baseFee = FlowFees.computeFees(inclusionEffort: 1.0, executionEffort: UFix64(executionEffort))
+            
+            return baseFee * self.priorityFeeMultipliers[priority]!
+        }
+
+        /// getNextIDAndIncrement returns the next ID and increments the ID counter
+        access(self) fun getNextIDAndIncrement(): UInt64 {
             let nextID = self.nextID
             self.nextID = self.nextID + 1
             return nextID
         }
 
+        /// Gets a struct containing all the configurable metadata
+        /// of the Scheduler resource
+        access(all) fun getConfigMetadata(): SchedulerConfig {
```

Comment:
Allows anyone to query the metadata that governs the callback scheduler contract